chirho/dynamical/handlers/dynamical.py

```diff
@@ -19,78 +15,36 @@
 
 
 class SimulatorEventLoop(Generic[T], pyro.poutine.messenger.Messenger):
-    # noinspection PyMethodMayBeStatic
     def _pyro_simulate(self, msg) -> None:
-        dynamics, initial_state, start_time, end_time = msg["args"]
+        dynamics, state, start_time, end_time = msg["args"]
         if "solver" in msg["kwargs"]:
             solver = msg["kwargs"]["solver"]
-        else:
-            # Early return to trigger `simulate` ValueError for not having a solver.
+        else:  # Early return to trigger `simulate` ValueError for not having a solver.
             return
 
-        # Initial values. These will be updated in the loop below.
-        span_start_state = initial_state
-        span_start_time = start_time
-
-        previous_terminal_interruptions: Tuple[Interruption, ...] = tuple()
-        interruption_counts: Dict[Interruption, int] = dict()
-
         # Simulate through the timespan, stopping at each interruption. This gives e.g. intervention handlers
         #  a chance to modify the state and/or dynamics before the next span is simulated.
-        while span_start_time < end_time:
-            # Block any interruption's application that wouldn't be the result of an interruption that ended the last
-            #  simulation.
+        while start_time < end_time:
             with pyro.poutine.messenger.block_messengers(
-                lambda m: isinstance(m, Interruption)
-                and m not in previous_terminal_interruptions
-            ):
-                dynamics, span_start_state = apply_interruptions(
-                    dynamics, span_start_state
-                )
-
-            # Block dynamic interventions that have triggered and applied more than the specified number of times.
-            # This will prevent them from percolating up to the simulate_to_interruption execution.
-            with pyro.poutine.messenger.block_messengers(
-                lambda m: (
-                    isinstance(m, DynamicInterruption)
-                    and m.max_applications <= interruption_counts.get(m, 0)
-                )
-                or isinstance(m, SimulatorEventLoop)
+                lambda m: m is self or (isinstance(m, Interruption) and m.used)
```

Comment:
I plan to further simplify and modify this logic in a followup PR that should also address #160, but this is a good intermediate point.